presto-hive/src/main/java/com/facebook/presto/hive/HiveManifestUtils.java

```diff
@@ -148,6 +150,18 @@ public static long getFileSize(Page statisticsPage, int position)
         return partitionMetadata.build();
     }
 
+    public static long getManifestSizeInBytes(ConnectorSession session, PartitionUpdate partitionUpdate)
+    {
+        if (isFileRenamingEnabled(session) && partitionUpdate.containsNumberedFileNames()) {
+            List<FileWriteInfo> fileWriteInfos = new ArrayList<>(partitionUpdate.getFileWriteInfos());
+            return compressFileNames(fileWriteInfos.stream().map(FileWriteInfo::getWriteFileName).collect(toImmutableList())).length()
+                    + compressFileSizes(fileWriteInfos.stream().map(FileWriteInfo::getFileSize).filter(Optional::isPresent).map(Optional::get).collect(toImmutableList())).length();
```

Comment:
This is duplicated logic. We can directly get the size info from `extraPartitionMetadata` in `HiveMetadata.finishInsert`. No need to compute it again.